src/btree/bt_cursor.c

```diff
@@ -553,25 +553,41 @@ __wt_btcur_search(WT_CURSOR_BTREE *cbt)
 
         if (btree->type == BTREE_ROW) {
             WT_ERR(__cursor_row_search(cbt, false, cbt->ref, &leaf_found));
-            if (leaf_found && cbt->compare == 0)
+            if (leaf_found && cbt->compare == 0) {
+                if (F_ISSET(&cbt->iface, WT_CURSTD_KEY_ONLY))
+                    WT_RET(__wt_key_return(cbt));
+                return (0);
```

Comment:
This is wrong return (0); should be under if (F_ISSET(&cbt->iface, WT_CURSTD_KEY_ONLY)) the if condition.